test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -583,8 +613,27 @@ private NodeAndClient buildNode(int nodeId, long seed, Settings settings, Versio
         return new NodeAndClient(name, node);
     }
 
-    private String buildNodeName(int id) {
-        return nodePrefix + id;
+    private String buildNodeName(int id, Settings settings) {
+        String prefix = nodePrefix;
+        prefix = prefix + getRoleSuffix(settings);
+        return prefix + id;
+    }
+
+    /** returns a suffix string based on the node role. If no explicit role is defined, the suffix will be empty */
+    private String getRoleSuffix(Settings settings) {
+        String suffix = "";
+        if (Node.NODE_MASTER_SETTING.exists(settings) && Node.NODE_MASTER_SETTING.get(settings)) {
+            suffix = suffix + "m";
```

Comment:
I changed to use getAbbreviation but it's a little ugly because coord -only node doesn't have an abbreviation, so we get:

```
        if (Node.NODE_MASTER_SETTING.exists(settings) && Node.NODE_MASTER_SETTING.get(settings)) {
            suffix = suffix + DiscoveryNode.Role.MASTER.getAbbreviation();
        }
        if (Node.NODE_DATA_SETTING.exists(settings) && Node.NODE_DATA_SETTING.get(settings)) {
            suffix = suffix + DiscoveryNode.Role.DATA.getAbbreviation();
        }
        if (Node.NODE_MASTER_SETTING.exists(settings) && Node.NODE_MASTER_SETTING.get(settings) == false &&
            Node.NODE_DATA_SETTING.exists(settings) && Node.NODE_DATA_SETTING.get(settings) == false
            ) {
            suffix = suffix + "c";
        }
```

Let me know if you don't like it and I'll roll back.

Re ingest - I can see how it's useful to see the node roles reflected in the node name, but since I used the same suffix for data paths, I think it will be confusing (I restarted a node and just enabled ingest on it, why can't I find my data?). Again, I opted the simplest thing possible and didn't make a separate prefix for name and path.